lib/ansible/modules/cloud/hcloud/hcloud_ssh_key.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_ssh_key
+
+short_description: Create and manage ssh keys on the Hetzner Cloud.
+
+version_added: "2.8"
+
+description:
+    - Create, update and manage ssh keys on the Hetzner Cloud.
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the Hetzner Cloud ssh_key to manage.
+            - Only required if no ssh_key I(name) is given
+        type: int
+    name:
+        description:
+            - The Name of the Hetzner Cloud ssh_key to manage.
+            - Only required if no ssh_key I(id) is given or a ssh_key does not exists.
```

Comment:
i would suggest to make `name` required if `state=present` because idempotency: you should not known the state (already exists or not)

in arg spec
~~~python
required_if=[[ 'state', 'present', ['name']],
~~~